backend/LeetWars.Core/LeetWars.Core.DAL/Interfaces/IUidProvider.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace LeetWars.Core.DAL.Interfaces
+{
+    public interface IUidProvider
```

Comment:
caanot we reuse our existing UserProvider?

Reply:
Our existing provider's implementation in our app depends on the dbcontext to be already initialized, creating a circular dependency

Reply:
Can we rework it to work in your way? we need to combine them as for now we are having 2 providers for the same thing